Source/Core/InputCommon/ControllerEmu/ControlGroup/Cursor.cpp

```diff
@@ -71,80 +79,82 @@ Cursor::ReshapeData Cursor::GetReshapableState(bool adjusted)
   const ControlState y = controls[0]->GetState() - controls[1]->GetState();
   const ControlState x = controls[3]->GetState() - controls[2]->GetState();
 
-  // Return raw values. (used in UI)
+  // Return raw values
   if (!adjusted)
     return {x, y};
 
-  return Reshape(x, y, 0.0);
+  return Reshape(x, y, 0.0, std::numeric_limits<ControlState>::infinity());
 }
 
 ControlState Cursor::GetGateRadiusAtAngle(double ang) const
 {
   return SquareStickGate(1.0).GetRadiusAtAngle(ang);
 }
 
-Cursor::StateData Cursor::GetState(const bool adjusted)
+Cursor::StateData Cursor::GetState(float absolute_time_elapsed, bool is_ui)
 {
-  if (!adjusted)
-  {
-    const auto raw_input = GetReshapableState(false);
-
-    return {raw_input.x, raw_input.y};
-  }
-
   const auto input = GetReshapableState(true);
 
-  // TODO: Using system time is ugly.
+  int i = is_ui ? 1 : 0;
+
   // Kill this after state is moved into wiimote rather than this class.
   const auto now = Clock::now();
   const auto ms_since_update =
-      std::chrono::duration_cast<std::chrono::milliseconds>(now - m_last_update).count();
-  m_last_update = now;
-
-  const double max_step = STEP_PER_SEC / 1000.0 * ms_since_update;
+      std::chrono::duration_cast<std::chrono::microseconds>(now - m_last_update[i]).count() /
+      1000.0;
+  m_last_update[i] = now;
 
   // Relative input:
   if (m_relative_setting.GetValue() ^ (controls[6]->GetState<bool>()))
   {
     // Recenter:
     if (controls[5]->GetState<bool>())
     {
-      m_state.x = 0.0;
-      m_state.y = 0.0;
+      m_state[i].x = 0.0;
+      m_state[i].y = 0.0;
     }
     else
     {
-      m_state.x = std::clamp(m_state.x + input.x * max_step, -1.0, 1.0);
-      m_state.y = std::clamp(m_state.y + input.y * max_step, -1.0, 1.0);
+      // If we are using a mouse axis to drive the cursor (there are reasons to), we want the
+      // step to be independent from the game speed, otherwise it would move at a different speed
+      // depending on the game speed.
+      // In other words, we want to be indipendent from time, and just use it as an absolute cursor.
+      // Of course if the emulation can't keep up with the wii mote
+      // emulation rate, the absolute time won't be accurate.
+      bool use_absolute_time = m_relative_absolute_time_setting.GetValue() && !is_ui;
```

Comment:
I also thought about this when noticing poor behavior with relative mouse mapped to relative pointing.

Instead of adding this option I think we can just always use the new "time elapsed" argument.

Users are already confused enough when mapping pointing. I really don't want to add another relative input related checkbox.

Reply:
Well the tooltip explains it, and I do like the choice as it depends on what you want to do. You might not always want it on depending on the relative input source.

Reply:
I really don't think the old timing is needed. The new timing might require tweaked "range" settings or whatever to restore old behavior but after that it should only make a difference when the game isn't running at 100% speed.

The advanced user want's their mouse movement of one pixel to do the same thing every time. (new timing)
The average user just want a dirty way to move the "cursor" around the screen with an analog stick and they don't care.

Reply:
It does matter if you are using an analog stick to drive the relative wiimote cursor. In that case you don't want to use that setting to actually have the cursor to move independently from the game speed. That's why I made it off by default.

Reply:
Yes I confirm what I said above, when using an analog stick to drive the relative wiimote cursor, you want the absolute time setting to be off, I tested it.
Also relative mouse movement gets much more precise with this setting as the chrono time delta was extremely variable between frames.

Reply:
m_relative_absolute_time_setting won't be needed anymore given I have made an input function to do the same thing (per mapping). It will be in a future PR but at least this one will be smaller.